src/tsickle.ts

```diff
@@ -888,6 +888,36 @@ class ExternsWriter extends ClosureRewriter {
       this.writeExternsFunction(name.getText(), paramNames, namespace);
     }
 
+    // Handle method declarations/signatures first, since we need to deal with overloads.
+    // TODO: Build a collection of the remaining members
+    let methods: {[id: string]: ts.MethodDeclaration[];} = {};
+    for (let member of decl.members) {
+      if (member.kind === ts.SyntaxKind.MethodSignature ||
+          member.kind === ts.SyntaxKind.MethodDeclaration) {
```

Comment:
I changed this in the follow up commit. The Signature|Declaration now are added to a separate array when we walk the full list of members. However, I missed removing this recheck. Another commit is coming.